### PR TITLE
Allow passing algorithm for token

### PR DESCRIPTION
**Pull request by fabbazon**
## Hey, I just made a Pull Request!

Work was done to add the ability to setup a custom signing key algorithm when generating JWT tokens. 

This PR allows changing the JWT token signature algorithm when it's generated in the authentication backend.

#### :heavy_check_mark: Checklist

- [X ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by benjdlambert** [comment]
Hey :wave: Why do you need this? Whats the use case? Just a little nod to some of the work that we're currently looking at which is part of the https://github.com/backstage/backstage/milestone/36, which is to standardize this header, and possibly open up the ability to override the generation of the token into whatever method you want which I think would solve this feature too.

**Comment by fabbazon** [comment]
Not all the JWT verifiers support ES256. For example the AWS API gateway currently supports only RS256. With this change I can use the backstage generated token with Backstage deployed in Fargate with an API gateway in front of it. 

  I believe it's an easy interim solution while the work on the new header system is done.

File: plugins/auth-backend/api-report.md [hunk 1]
```diff
@@ -650,6 +650,8 @@ export interface RouterOptions {
   // (undocumented)
   providerFactories?: ProviderFactories;
   // (undocumented)
+  tokenKeyAlgorithm?: string;
```
Comment (benjdlambert) [inline plugins/auth-backend/api-report.md:653]: Maybe we can change this name to `tokenFactoryAlgorithm` instead as it's being passed to the `tokenFactory`?

File: .changeset/plenty-clouds-guess.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-auth-backend': minor
```
Comment (benjdlambert) [inline .changeset/plenty-clouds-guess.md:2]: Think that this should be a patch change as the `auth-backend` is <1.x

**Review by benjdlambert — COMMENTED** [review]
Just some small nits :pray:

File: .changeset/plenty-clouds-guess.md (hunk 2 above)
Comment (fabbazon) [reply .changeset/plenty-clouds-guess.md]: Good call!

**Review by benjdlambert — APPROVED** [review]
Great! Looks good to me! :pray:

**Pull request merged**